shadowsocks-csharp/View/ConfigForm.cs

```diff
@@ -273,16 +273,16 @@ private void AddButton_Click(object sender, EventArgs e)
             {
                 return;
             }
-            Server server = Configuration.GetDefaultServer();
-            _modifiedConfiguration.configs.Add(server);
+            Configuration.AddDefaultServerOrServer(_modifiedConfiguration);
             LoadServerNameListToUI(_modifiedConfiguration);
             ServersListBox.SelectedIndex = _modifiedConfiguration.configs.Count - 1;
             _lastSelectedIndex = ServersListBox.SelectedIndex;
         }
 
         private void DuplicateButton_Click(object sender, EventArgs e)
         {
-            if (!ValidateAndSaveSelectedServerDetails())
+            if (_lastSelectedIndex == -1 || _lastSelectedIndex > _modifiedConfiguration.configs.Count

```

Comment:
Here you can consider the original code, because the `DeleteButton_Click` method guarantees that `_lastSelectedIndex` is at least 0 instead of -1.